### ISSUE TITLE
Expired/disabled bans imported erroneously

### ISSUE THREAD

**Issue by jim-p** [a5mith/nodebb-plugin-import-smf#9]
We have a rather large userbase (considering banned users, close to 300k users) that mostly imported OK from SMF but we hand a handful of users (43 users, ~0.015%) that had incorrect bans. After digging around in the old SMF ban tables we found that a few conditions which led to SMF users without an active ban ending up banned in NodeBB after import. These include:

* Manually disabled ban entries: Items from smf_ban_groups where expire_time = 0
  `select * from smf_ban_groups where expire_time = 0;`
* Bans with an expiration date, need to check if current time > expire_time, or skip import: Items from smf_ban_groups where expire_time <> 0 and expire_time is not null
  `select * from smf_ban_groups where expire_time <> 0 and expire_time is not null;`
* Items from smf_ban_groups with no corresponding ban triggers in smf_ban_items (Selecting from smf_ban_items where smf_ban_items.id_ban_group = smf_ban_groups.id_ban_group ends up with an empty set)
  `select * from smf_ban_groups where id_ban_group not in (select id_ban_group from smf_ban_items);`

**Comment by akhoury** [comment]
it seems you're well aware of the solution, I would really appreciate a pull request 😄

**Comment by jim-p** [comment]
Fair enough! I'd love to dig in and try that but at the moment I don't have time available to take that on. :)

Now that our import is done we likely won't need this again, but I figured it was worth noting for the next person who came along and needed it.
